86BoxManager/Settings/Settings.cs

```diff
@@ -0,0 +1,276 @@
+                    EnableGridLines = Convert.ToBoolean(regkey.GetValue("EnableGridLines"));
+                }
+
+                regkey.Close();
+            }
+            catch (Exception ex)
+            {
+#if DEBUG
+                MessageBox.Show($"Exception! {ex.Message}\n\n{ex.StackTrace}");
+#else
+                MessageBox.Show("An error has occurred. Please provide the following information to the developer:\n" + ex.Message + "\n" + ex.StackTrace, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+#endif
+                RestoreToDefaults(false); 
+            }
+        }
+        
+        /// <summary>
+        /// Restore to defaults
+        /// </summary>
+        public static void RestoreToDefaults(bool CreateSettings)
+        {
+            //Create the key and reopen it for write access
+
+            if (CreateSettings)
+            {
+                if (!CreateSettingsKey())
+                {
+                    MessageBox.Show("An error occurred creating settings.", "Warning!", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+                }
+            }
+            
+            CFGDir = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile) + @"\86Box VMs\";
+            EXEDir = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86) + @"\86Box\";
+            MinimizeOnVMStart = false;
+            ShowConsole = true;
+            MinimizeToTray = false;
+            CloseToTray = false;
+            EnableLogging = false;
+            LaunchTimeout = "5000";
+            EnableGridLines = false;
+            LogPath = "";
+        }
+
+        /// <summary>
+        /// Create the settings keys.
+        /// </summary>
+        /// <returns>BOOL - success code</returns>
+        private static bool CreateSettingsKey()
+        {
+            try
+            {
+                RegistryKey RegKey = Registry.CurrentUser.CreateSubKey(@"SOFTWARE\86Box");
+
+                // 2020-11-23   Connor Hyde (starfrost) add additional error checking
+
+                if (RegKey == null) return false;
+
+                RegKey = Registry.CurrentUser.OpenSubKey(@"SOFTWARE\86Box", true);
+
+                if (RegKey == null) return false;
+
+                RegKey.CreateSubKey("Virtual Machines");
+                return true; 
+            }
+            catch (ArgumentNullException err)
+            {
+#if DEBUG
+                MessageBox.Show($"86Box Manager [Debug] ArgumentNullException occurred in Settings.CreateSettingsKey() [static] (bug?): {err.Message} \n\n{err.StackTrace}");
+#endif
+                return false;
+            }
+            catch (IOException err)
+            {
+#if DEBUG
+                MessageBox.Show($"86Box Manager [Debug] IOException occurred in Settings.CreateSettingsKey() [static]: {err.Message} \n\n{err.StackTrace}");
+#endif
+                return false;
+            }
+        }
+
+        public static bool SaveSettings()
```

Comment:
Settings aren't actually being saved on my end.

Reply:
Most settings were not saved due to the fact that setting values weren't being updated internally. This has been fixed